apps/core/components/link/index.tsx

```diff
@@ -1,21 +1,56 @@
+'use client';
+
 // eslint-disable-next-line @typescript-eslint/no-restricted-imports
 import NextLink from 'next/link';
-import { ComponentPropsWithRef, ElementRef, forwardRef } from 'react';
+import { useRouter } from 'next/navigation';
+import { ComponentPropsWithRef, ElementRef, forwardRef, useReducer } from 'react';
 
 import { cn } from '~/lib/utils';
 
-type Props = ComponentPropsWithRef<typeof NextLink>;
+type NextLinkProps = Omit<ComponentPropsWithRef<typeof NextLink>, 'prefetch'>;
+
+interface PrefetchOptions {
+  prefetch?: 'hover' | 'viewport' | 'none';
+  prefetchKind?: 'auto' | 'full';
+}
+
+type Props = NextLinkProps & PrefetchOptions;
 
+/**
+ * This custom `Link` component extends Next.js's `Link` with additional prefetching controls, making
```

Comment:
It took me a bit to realize that `auto` and `full` were describing the difference between Next.js's `undefined` and `true` values, I think that could probably be better explained here. That said, this is probably fine for now as I think ultimately we're going to probably want to reference the catalyst.dev/docs here (as this kind of thing will certainly live there), which we should have linkable soon enough, so we can iterate there instead.